### PR TITLE
Feature/testing

### LINKED ISSUE #20: Add Cypress testing

### PR DESCRIPTION
**Pull request by anthony-iacono**
### Features
- Closes #20 

### Refactors 
- Add class selectors with `cy-` prefix to isolate from selectors used for other purposes

### Issues and Technical Debt
- None

### Other Comments, Concerns, or Questions
- None

### Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature work
- [x] New and existing unit tests pass locally with my changes
- [ ] I ran Lighthouse and WAVE accessibility audits and fixed errors/warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/Card/Card.tsx [hunk 1]
```diff
@@ -9,9 +9,9 @@ interface CardProps {
 const Card = ({ title, image }: CardProps): JSX.Element => {
   return (
     <div className="card-container">
-      <article className="news-card">
-        <img className="article-image" src={ image } alt={ title } />
-        <h2 className="article-title">{ title }</h2>
+      <article className="news-card cy-article-card">
+        <img className="article-image cy-article-image" src={ image } alt={ title } />
+        <h2 className="article-title cy-article-title">{ title }</h2>
```
Comment (najinl) [inline src/components/Card/Card.tsx:14]: Is this a "best practice" I don't know about?! The "cy" in front of the className? I'd love to more about it.

**Review by najinl — APPROVED** [review]
Thank you for adding the feed testing! Looks good. I'm having a hard time thinking of many sad paths, but a good one would be to add error handling testing. I'm happy to help out on that.

File: cypress/integration/form_spec.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  it('should render the appropriate elements', () => {
+    cy.contains('WellNews')
+      .get('.cy-sentiment-question').contains('how are you feeling today?')
+      .get('.cy-happy-btn').should('be.visible')
+      .get('.cy-neutral-btn').should('be.visible')
+      .get('.cy-sad-btn').should('be.visible')
+      .get('.cy-view-synopsis-button').should('be.visible')
+  })
+
+  it('should take you to the news feed when the happy sentiment button is chosen', () => {
```
Comment (nzambonivergara) [inline cypress/integration/form_spec.js:15]: Do these three tests need an intercept?

File: cypress/integration/Feed_spec.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+describe('Feed', () => {
+  beforeEach(() => {
+    cy.intercept("https://api.nytimes.com/svc/topstories/v2/home.json?api-key=GKUzDD1VY9ssjZ1AGusX3ci6AeoXCaSr", {fixture: 'nyt-article-fetch.json'})
```
Comment (nzambonivergara) [inline cypress/integration/Feed_spec.js:3]: I didn't know how to use fixtures... thank you for integrating them, @najinl! So cool!

**Review by nzambonivergara — APPROVED** [review]
Thanks @najinl and @anthony-iacono for working on these! I agree with Nadia that we can come up with some sad path testing and would also be happy to help integrate those.

File: cypress/integration/Feed_spec.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  })
+
+  it('Should show the header', () => {
+    cy.visit('/').get('.cy-sad-btn').click()
+      .get('.cy-header-text').contains('WellNews')
+  })
+
+  it('Should, when the happy button is clicked, show the articles sorted in ascending order by sentiment', () => {
+    cy.visit('/').get('.cy-happy-btn').click()
+      .get('.cy-article-card').first().contains('Should I Tell a Facebook Friend I Had an Affair With Her Partner?')
```
Comment (nzambonivergara) [inline cypress/integration/Feed_spec.js:15]: I never know how much testing is enough and usually I check all of the pieces of the part I am testing. For example, whether the src of the image is the one that I am expecting as well as the title. But I am not sure if it is necessary... maybe something I will ask during our stand up today.

File: src/components/Card/Card.tsx (hunk 1 above)
Comment (anthony-iacono) [reply src/components/Card/Card.tsx:14]: Yep, it's a best practice. See the Cypress [docs](https://docs.cypress.io/guides/references/best-practices#Selecting-Elements)

File: cypress/integration/form_spec.js (hunk 2 above)
Comment (anthony-iacono) [reply cypress/integration/form_spec.js:15]: You're right. I'll add one to help avoid rate limiting.

**Pull request merged**